File: snippets/single-file-single-thread/string-formatting/java/basic.java
String name = "Alice";
int age = 30;
double price = 19.99;

// String.format()
String message = String.format("Name: %s, Age: %d", name, age);
String formattedPrice = String.format("Price: $%.2f", price);

// printf style
System.out.printf("Name: %s, Age: %d%n", name, age);

File: snippets/single-file-single-thread/string-formatting/javascript/basic.js
const name = "Alice";
const age = 30;
const price = 19.99;

// Template literals (recommended)
const message = `Name: ${name}, Age: ${age}`;
const formattedPrice = `Price: $${price.toFixed(2)}`;

// Number formatting
const formatted = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD'
}).format(price);  // "$19.99"
